Cython/Compiler/ModuleNode.py

```diff
@@ -810,18 +810,48 @@ def generate_module_preamble(self, env, options, cimported_modules, metadata, co
         self._put_setup_code(code, "PythonCompatibility")
         self._put_setup_code(code, "MathInitCode")
 
-        if options.c_line_in_traceback:
-            cinfo = "%s = %s; " % (Naming.clineno_cname, Naming.line_c_macro)
-        else:
-            cinfo = ""
-        code.putln("#define __PYX_MARK_ERR_POS(f_index, lineno) \\")
+        # Error handling and position macros.
         # Using "(void)cname" to prevent "unused" warnings.
-        code.putln("    { %s = %s[f_index]; (void)%s; %s = lineno; (void)%s; %s (void)%s; }" % (
-            Naming.filename_cname, Naming.filetable_cname, Naming.filename_cname,
-            Naming.lineno_cname, Naming.lineno_cname,
-            cinfo,
-            Naming.clineno_cname,
-        ))
+        mark_errpos_code = (
+            "#define __PYX_MARK_ERR_POS(f_index, lineno)  {"
+            f" {Naming.filename_cname} = {Naming.filetable_cname}[f_index];"
+            f" (void) {Naming.filename_cname};"
+            f" {Naming.lineno_cname} = lineno;"
+            f" (void) {Naming.lineno_cname};"
+            "%s"  # for C line info
+            f" (void) {Naming.clineno_cname}; "  # always suppress warnings
+            "}"
+        )
+        cline_info = f" {Naming.clineno_cname} = {Naming.line_c_macro};"
+
+        # Show the C code line in tracebacks or not? C macros take precedence over (deprecated) options.
+        # 1)  "CYTHON_CLINE_IN_TRACEBACK=0"  always disables C lines in tracebacks
+        # 2) "CYTHON_CLINE_IN_TRACEBACK_RUNTIME=1"  enables the feature + runtime configuration
+        # 2a) "options.c_line_in_traceback=True"   changes the default to CYTHON_CLINE_IN_TRACEBACK_RUNTIME=1
+        # 2b) "options.c_line_in_traceback=False"  changes the default to disable C lines
+        # 4)  "CYTHON_CLINE_IN_TRACEBACK=1"  enables C lines without runtime configuration
+        # 5)  if nothing is set, the default is to disable the feature
```

Comment:
Looks like you wanted this to feel like a table, so I'll push some spacings around.
```suggestion
        # 1) "CYTHON_CLINE_IN_TRACEBACK=0"  always disables C lines in tracebacks
        # 2) "CYTHON_CLINE_IN_TRACEBACK_RUNTIME=1" enables the feature + runtime configuration
        # 2a) "options.c_line_in_traceback=True"   changes the default to CYTHON_CLINE_IN_TRACEBACK_RUNTIME=1
        # 2b) "options.c_line_in_traceback=False"  changes the default to disable C lines
        # 4) "CYTHON_CLINE_IN_TRACEBACK=1"         enables C lines without runtime configuration
        # 5) if nothing is set, the default is to disable the feature
```